src/NBitcoin/ConcurrentChain.cs

```diff
@@ -26,15 +26,16 @@ public ConcurrentChain()
 
         public ConcurrentChain(BlockHeader genesisHeader)
         {
+            // TODO: Pass NetworkOptions to GetHash
             this.SetTip(new ChainedBlock(genesisHeader, genesisHeader.GetHash(), 0));
         }
 
         public ConcurrentChain(Network network)
         {
             if (network != null)
             {
-                Block genesis = network.GetGenesis();
-                this.SetTip(new ChainedBlock(genesis.Header, genesis.Header.GetHash(), 0));
+                Block genesis = network.GetGenesis();                
```

Comment:
Ah I see thats already the case so we can create a local Network variable

Reply:
Added.